client/libra-dev/src/transaction.rs

```
@@ -0,0 +1,633 @@
+                        addr_buffer.copy_from_slice(addr.as_ref());
+                        address = Some(addr_buffer);
+                    }
+                    _ => {}
+                });
+                if let (Some(val), Some(add)) = (value, address) {
+                    txn_payload = Some(LibraTransactionPayload {
+                        txn_type: TransactionType::PeerToPeer,
+                        args: LibraP2PTransferTransactionArgument {
+                            value: val,
+                            address: add,
+                        },
+                    });
+                } else {
+                    return LibraStatus::InternalError;
+                }
+            }
+            "mint_transaction" => {
+                let args = script.args();
+                let mut value = None;
+                let mut address = None;
+                args.iter().for_each(|txn_arg| match txn_arg {
+                    TransactionArgument::U64(val) => {
+                        value = Some(*val);
+                    }
+                    TransactionArgument::Address(addr) => {
+                        let mut addr_buffer = [0u8; ADDRESS_LENGTH];
+                        addr_buffer.copy_from_slice(addr.as_ref());
+                        address = Some(addr_buffer);
+                    }
+                    _ => {}
+                });
+                if let (Some(val), Some(add)) = (value, address) {
+                    txn_payload = Some(LibraTransactionPayload {
+                        txn_type: TransactionType::Mint,
+                        args: LibraP2PTransferTransactionArgument {
+                            value: val,
+                            address: add,
+                        },
+                    });
+                } else {
+                    return LibraStatus::InternalError;
+                }
+            }
+            &_ => {
+                // support other transactions
```

Should this return an error of some kind instead of silently proceeding?